Send forgot password email by language type

absis opened the pull request.

Commit:
Send forgot password email by language type.

I've added new required param lang. When user calls forgotPassword
have to pass this params for send the correct language email.

Commit:
Return new code error for missing or invalid lang param

Commit:
Upgraded package.json version to 0.0.384

louis-sanchez committed:
[Fix] pass domain to sign the card

· Because of a change inside lib eyeos-auth, now we need to pass the
domain whenever we sign the card.

· fixed test

Commit:
Upgraded package.json version to 0.0.385

afiestas commented on authentication_v2/src/lib/forgotPassword/ForgotPassword.js:
```diff
@@ -31,13 +31,19 @@ ForgotPassword.prototype.postForgot = function(req, res) {
     var responseOkMessage = "We've sent an email with the instructions for recover your password";
 
     userData.username = req.body.username;
+    userData.lang = req.body.lang;
```
Would be nice to control more this input, we can't trust what we are given by the client and this var is used to build a path later on.

absis replied:
Done in commit 0fcd012adf4e7de2005d7d67cc7dce627e3c330a

Commit:
Check request param lang.

I've created a new array in settings with supported languages.
I've checked if the lang param is in this array.

Commit:
Upgraded package.json version to 0.0.386

Commit:
Add new param for request forgotPassword: domain.

This param is used for create the url sended in the forgot password
mail instead of get the domain from the username.

Commit:
Upgraded package.json version to 0.0.387

afiestas committed:
Sort dependencies correctly between Dev and Production

Basically some modules that belong to dev where in production and the
other way around

afiestas committed:
Remove replaceshell dependency

Not used according to git-grep, and test pass

afiestas committed:
Remove shelljs dependency

test pass and git grep yields no results

afiestas committed:
Unhardcore version of eyeos-service-bus

CI!

afiestas committed:
node-uuid is actually a test dependency

afiestas committed:
Removed authentication-installed

No longer needed, it has not been since the RPM era

afiestas committed:
Port to alpine6-node-base

The usual, install dev dependencies and a few runtime ones specific from
authentication like cairo and ttf-liberation for the captcha.

afiestas committed:
set InstallationDir to var/service

Got removed with the porting, Whoops!

afiestas committed:
Add force to rm in Dockerfile

Just in case it fails

afiestas committed:
Install bash for the build process

afiestas committed:
Add krb5-dev to Dockerfile

For some reason build is not breaking locally but it does in the CI :/
it seems that it is missing gasspi.h which is located in krb5 package.

afiestas committed:
Add final missing dependencies

Dockerfiles like this a re mess, we have to find a better way.

afiestas committed:
Let's use the bash available in alpine6-node-base image

Ash should work, if not we can always install bash again for the build
process.

afiestas committed:
Remove only installed packages instead of dependencies

Apk takes care of removing dependencies for us, this way Dockerfile is
simpler and corrrect.

Commit:
Add called install and delete common packages

Commit:
Install/Uninstall extra packages injected as parameters scripts-base

Commit:
Called to script install/purgue change

Commit:
Generate list of dependencies to install/purgue into Dockerfile

Commit:
Added information about variable common

Commit:
add .dockerignore

Using this our images will be thinner because they won't add those
files/folders.

Commit:
do not copy alpine-*.list when they are already being copied

They are being copied by the COPY . $InstallationDir sentence.
